.buildkite/pipelines/coverage_converter_pipeline.yml

```diff
@@ -13,22 +18,19 @@ steps:
       - cd metrics/coverage_converter
       - pub get
       - pub run test_coverage --print-test-output --no-badge
-    artifact_paths: "metrics/coverage_converter/coverage/lcov.info"
-
-  - label: "Export coverage report"
-    key: "coverage"
-    depends_on: "tests"
-    commands:
-      - curl -o coverage_converter -k https://github.com/platform-platform/monorepo/releases/download/coverage-converter-snapshot/coverage_converter_linux -L
-      - chmod a+x coverage_converter
-      - buildkite-agent artifact download metrics/coverage_converter/coverage/lcov.info .
-      - ./coverage_converter lcov -i metrics/coverage_converter/coverage/lcov.info -o coverage-summary.json
-    artifact_paths: "coverage-summary.json"
+      - coverage_converter lcov -i coverage/lcov.info -o coverage-summary.json
```

Comment:
It seems like separating the `tests` and `coverage` steps is more granular and clear though more complex in implementation. The main idea is that coverage exporting failure is not the same as test failure since there can be an error related to the `coverage_converter` tool. Thus, separating these steps allows us to detect the reason for build failure easier and faster.